sdk/include/opentelemetry/sdk/trace/recordable.h

```diff
@@ -94,6 +120,19 @@ class Recordable
    * @param duration the duration to set
    */
   virtual void SetDuration(std::chrono::nanoseconds duration) noexcept = 0;
+
+private:
+  /**
+   * Maintain a static empty map that represents default attributes.
+   * Avoid constructing a new empty map everytime a call is made with default attributes.
+   */
+  static const nostd::shared_ptr<trace_api::KeyValueIterableView<std::map<std::string, int>>>
+      &GetDefaultAttributes() noexcept
+  {
+    static const nostd::shared_ptr<trace_api::KeyValueIterableView<std::map<std::string, int>>>
+        kDefaultAttributes({});
+    return kDefaultAttributes;
```

Comment:
I think there's no need for a `shared_ptr`. You could just return a `const trace_api::KeyValueIterable &`.